### PR TITLE
Feature/reg 1327 cucumber sampling with table data

### PR DESCRIPTION
**Pull request by ameenhaq**

### CODE REVIEW

File: resources/features/Sampling.feature [hunk 1]
```diff
@@ -1,73 +1,146 @@
...
+        And a Strata of selection type Prn-Sampling from Stratification Properties file:
+            | inqcode | cell_no | cell_desc | seltype | lower_class | upper_class | lower_size | upper_size | prn_start   | no_reqd |
+            | 687     | 5816    | Sample    | P       | 45111       | 45190       | 50         | 9          | 0.539298879 | 3       |
+            | 687     | 5811    | Admin     | U       | 45111       | 45190       | 0          | 9          | 0.000000000 | 0       |
+            | 687     | 5812    | Admin     | U       | 45111       | 45190       | 10         | 49         | 0.000000000 | 0       |
         When a <language> Sample is created from a Stratified Frame
-        Then a Sample DataFrame is returned and exported to CSV with the inclusion of stratas with outbound Sample Size parameter
+        Then a Sample is returned and exported to CSV with the inclusion of stratas with outbound Sample Size parameter
```
Comment (nigelhp) [inline resources/features/Sampling.feature]: "out of bounds" as opposed to "outbound"?

File: resources/features/Sampling.feature [hunk 2]
```diff
@@ -81,17 +154,39 @@ Feature: Creating a Sample
...
+            | 1100000011 | 9906000415 | &BAMCFLINT                    |              | 1 GARENDON WAY           | GROBY          | LEICESTER   |                         |          | LE6 0YR  | 1           | 46160 | 1           | 0         | 400          | 400          | 0            | 0             | 0            | 0.269071541 | 5819    |
+        And a Strata of selection type Prn-Sampling from Stratification Properties file:
+            | inqcode | cell_no | cell_desc | seltype | lower_class | upper_class | lower_size | upper_size | prn_start   | no_reqd |
+            | 687     | 5819    | Sample    | P       | 45111       | 45190       | 50         | 9          | 0.129177704 | 100000  |
         When a <language> Sample is created from a Stratified Frame
-        Then a Sample DataFrame is returned and exported to CSV with the invalid Sample Size strata logged and entire Frame returned for that strata
+        Then a Sample is returned and exported to CSV with the strata containing an invalid Sample Size is replaced with the population size and is logged
```
Comment (nigelhp) [inline resources/features/Sampling.feature]: are we actually asserting on logging at this time?

File: .gitignore [hunk 3]
```diff
@@ -29,7 +29,8 @@ R/NAMESPACE
 R/.Rapp.history
 feature-REG-1084.Rproj
 metastore_db
-resources/outputs/sampling/*
+resources/outputs/sampling
```
Comment (nigelhp) [inline .gitignore]: are these ignores still required?

File: resources/features/Sampling.feature (hunk 2 above)
Comment (pwetherill) [inline resources/features/Sampling.feature]: Recommend changing wording for clarity to:
Then a Sample containing the total population in the strata is returned and exported to CSV with an error statting target Sample Size is greater than strata size is logged

File: resources/features/Sampling.feature [hunk 4]
```diff
@@ -1,73 +1,146 @@
...
-
-    Scenario Outline: Stratas contain Sample Size parameter that are out of bounds of the sorted Frame records.
-        Given a Stratified Frame and a stratification properties file paths:
-            | data_input_path   | strat_properties        | expected_output   |
-            | <input_path>      | <strat_properties_path> | <output_path>     |
+        | language |
+        | Scala    |
+
+
+    Scenario Outline: Strata contains a Sample Size parameter that is out of bounds of the sorted Stratified Frame population.
```
Comment (pwetherill) [inline resources/features/Sampling.feature]: Test is for PRN 'wrapping' - The Given, When, Then is the same as a normal PRN selection as you should still get a full output. Or do we add an additional Then statement that reads something like:
"the Sample continues to select the Sample Size restarting from the begining of the PRN line"?

Amend Scenario Outline to read:
Strata contains a Sample Size parameter that exceeds the end of the sorted PRN line.
Amend the Then statement to read:
a Sample containing the Sample Size from the PRN-sampling strata is returned and exported to CSV

File: resources/features/Sampling.feature [hunk 5]
```diff
@@ -1,73 +1,146 @@
...
+        And a Strata of selection type Prn-Sampling from Stratification Properties file:
+            | inqcode | cell_no | cell_desc | seltype | lower_class | upper_class | lower_size | upper_size | prn_start   | no_reqd |
+            | 687     | 5813    | Sample    | P       | 45111       | 45190       | 50         | 9          | 0.129110904 | 4       |
+            | 687     | 5811    | Admin     | U       | 45111       | 45190       | 0          | 9          | 0.000000000 | 0       |
+            | 687     | 5812    | Admin     | U       | 45111       | 45190       | 10         | 49         | 0.000000000 | 0       |
         When a <language> Sample is created from a Stratified Frame
-        Then a Sample DataFrame containing the Prn-Sampling strata is returned and exported to CSV
+        Then a Sample containing the sample selection from the Prn-Sample strata is returned and exported to CSV
```
Comment (pwetherill) [inline resources/features/Sampling.feature]: Should "sample selection" be changed to "Sample Size" as a means of explicitly checking we have the right return?
If so, amendment should be reflected across the tests.

**Pull request merged**